esphome/components/animation/__init__.py

```diff
@@ -48,6 +59,13 @@ def validate_cross_dependencies(config):
             # Not setting default here on purpose; the default depends on the image type,
             # and thus will be set in the "validate_cross_dependencies" validator.
             cv.Optional(CONF_USE_TRANSPARENCY): cv.boolean,
+            cv.Optional(CONF_LOOP): cv.All(
+                {
+                    cv.Optional(CONF_START_FRAME, default=0): cv.int_,
+                    cv.Optional(CONF_END_FRAME): cv.Any(None, cv.int_),
+                    cv.Optional(CONF_REPEAT): cv.Any(None, cv.int_),
```

Comment:
Do you need `None` here at all if they are optional anyway?

Reply:
^^^ second this -- I just tried locally and this seems to work whether the parameter is specified or not:

```suggestion
                    cv.Optional(CONF_END_FRAME): cv.int_,
                    cv.Optional(CONF_REPEAT): cv.int_,
```

Reply:
I followed the upstream voluptuous documentation, but you are right; that is superfluous.

Reply:
I should've mentioned this the first time, but `cv.positive_int` would probably be even better for these (I'm assuming you'd never want to allow a value less than zero). 😄

Reply:
Left a suggestion in case it's easier to just merge in the change that way. Sorry for not catching this the first time!